File: 7KYU/Going to the cinema/solution.clj
(ns movie.core)

(defn get-total [card ticket discount n]
  (let [a1 (* ticket discount)]
    (+ card (* a1 (/ (- 1 (Math/pow discount n))
                     (- 1 discount))))))

(defn movie [card ticket discount]
  (loop [count 1]
    (let [a (* count ticket)
          b (get-total card ticket discount count)]
      (if (> (- a b) 1)
        (do
          (int count))
        (recur (inc count))))))

File: 7KYU/Going to the cinema/solution.go
package kata

import "math"

func getTotal(card, ticket int, perc float64, cnt int) float64 {
	d := float64(ticket) * perc
	return float64(card) + d*(1-(math.Pow(perc, float64(cnt))))/(1.0-perc)
}

func Movie(card, ticket int, perc float64) int {
	cnt := 1

	for {
		a := cnt * ticket
		b := getTotal(card, ticket, perc, cnt)

		if (float64(a) - float64(b)) > 1 {
			return cnt
		} else {
			cnt += 1
		}
	}
}
